### PR TITLE
[refactor] deindent tuple parameter expansion

### PR DESCRIPTION
**Pull request by thewilsonator**

### CODE REVIEW

**Comment by thewilsonator** [comment]
Best reviewed with whitespace off.

File: src/dmd/semantic3.d [hunk 1]
```diff
@@ -479,37 +479,34 @@ private extern(C++) final class Semantic3Visitor : Visitor
 
             // Declare the tuple symbols and put them in the symbol table,
             // but not in parameters[].
-            if (f.parameterList.parameters)
+            if (f.parameterList.parameters) foreach (fparam; *f.parameterList.parameters)
```
Comment (Geod24) [inline src/dmd/semantic3.d]: That `for` => `foreach`  transformation makes me a bit anxious, but according to the comment, it's valid 🤷‍♂ 
However I think we  usually  do:
```suggestion
            if (f.parameterList.parameters)
            foreach (fparam; *f.parameterList.parameters)
```

which IMO makes more  sense as the leftmost item is the most visible, and we want `foreach` to be more visible than a trivial `null` check.

**Review by Geod24 — COMMENTED** [review]
Minor comments. You could also apply the foreach on L494 since you're already doing that transformation on L483.

**Comment by thewilsonator** [comment]
Done.

File: src/dmd/semantic3.d (hunk 1 above)
Comment (thewilsonator) [reply src/dmd/semantic3.d]: Done.

**Comment by thewilsonator** [comment]
No because the index is used over two different arrays. (apologies for the confusing done comments)

**Comment by Geod24** [comment]
> No because the index is used over two different arrays. (apologies for the confusing done comments)

I meant `foreach (j, 0 .. dim)`

**Comment by thewilsonator** [comment]
Done. Should be good to squash and merge.

**Comment by Geod24** [comment]
Noice

**Review by Geod24 — APPROVED** [review]

**Pull request merged**